### DISCUSSION TITLE
@bot.listen() not doing anything

### DISCUSSION THREAD

**Discussion by mrracoonman** [Rapptz/discord.py#6367] [Q&A] [answered]
I'm running this code, and its supposed to delete what a user says and have my bot say it with certain stuff around it.

```py
@bot.listen()
async def on_message(message):
    channel = bot.get_channel(channel I want)
    author = message.author
    if author == user id:
        await channel.delete_message(message)
        await channel.send(f"A Message!?,\n"
                           f"It reads: {message.content}")
    else:
        return
```

and its not throwing any errors, and when that user says something, the bot does nothing at all.
Could someone explain what I'm doing wrong?

Win 10, py 3.8

**Comment by Rapptz** [comment, marked as answer]
There are a few issues with this code as-is.

1. `author` is of type `discord.Member` (or `discord.User`). You cannot compare this to a numeric ID. If you want to compare it with an ID you need to do `author.id == 123456...`
2. There is no method called `delete_message` in any channel class.

**Comment by Cyanomous** [comment]
Here: https://stackoverflow.com/questions/42182243/deleting-user-messages-in-discord-py

**Comment by Cyanomous** [comment]
also, I think it might be `message.delete`

**Comment by bandwiches** [comment]
I think you're looking for [discord.TextChannel.delete_messages](https://discordpy.readthedocs.io/en/latest/api.html?highlight=delete_message#discord.TextChannel.delete_messages).
